server/persistence/dynamo/dynamo_service_interaction.go

```diff
@@ -0,0 +1,257 @@
+//nolint:dupl
+package dynamo
+
+import (
+	"context"
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+
+	"github.com/joshprzybyszewski/cribbage/model"
+	"github.com/joshprzybyszewski/cribbage/server/interaction"
+	"github.com/joshprzybyszewski/cribbage/server/persistence"
+)
+
+var _ persistence.InteractionService = (*interactionService)(nil)
+
+type interactionService struct {
+	ctx context.Context
+
+	svc *dynamodb.Client
+}
+
+func getInteractionService(
+	ctx context.Context,
+	svc *dynamodb.Client,
+) persistence.InteractionService {
+	return &interactionService{
+		ctx: ctx,
+		svc: svc,
+	}
+}
+
+func (is *interactionService) Get(
+	id model.PlayerID,
+) (interaction.PlayerMeans, error) {
+	ret := interaction.PlayerMeans{
+		PlayerID:      id,
+		PreferredMode: interaction.Unknown,
+	}
+
+	pkName := `:pID`
+	pk := string(id)
+	skName := `:sk`
+	sk := getSortKeyPrefix(is)
+	keyCondExpr := getConditionExpression(equalsID, pkName, hasPrefix, skName)
+
+	createQuery := func() *dynamodb.QueryInput {
+		return &dynamodb.QueryInput{
+			TableName:              aws.String(dbName),
+			KeyConditionExpression: &keyCondExpr,
+			ExpressionAttributeValues: map[string]types.AttributeValue{
+				pkName: &types.AttributeValueMemberS{
+					Value: pk,
+				},
+				skName: &types.AttributeValueMemberS{
+					Value: sk,
+				},
+			},
+		}
+	}
+	items, err := fullQuery(is.ctx, is.svc, createQuery)
+	if err != nil {
+		return interaction.PlayerMeans{}, err
+	}
+	err = is.populatePlayerMeansFromItems(&ret, items)
```

Comment:
I recently learned about the importance of pointer vs. value semantics. Why are we okay with `interaction.PlayerMeans` being defined everywhere with value semantics except for here?

Reply:
great question. I think I can rework it...